src/lib/ngsi/Scope.cpp

```diff
@@ -281,7 +309,31 @@ int Scope::fill
       pointV.clear();
       return -1;
     }
-    box.fill(pointV[0], pointV[1]);
+
+    // Check that points are different an not aligned (either horizontally or vertically)
+    if ((pointV[0]->latitude() == pointV[1]->latitude())  || (pointV[0]->longitude() == pointV[1]->longitude()))
+    {
+      *errorStringP = "box coordiantes are not defining an actual box";
+      pointVectorRelease(pointV);
+      pointV.clear();
+      return -1;
+    }
+
+    double minLat = (pointV[0]->latitude()  < pointV[1]->latitude())?  pointV[0]->latitude()  : pointV[1]->latitude();
+    double maxLat = (pointV[0]->latitude()  > pointV[1]->latitude())?  pointV[0]->latitude()  : pointV[1]->latitude();
+    double minLon = (pointV[0]->longitude() < pointV[1]->longitude())? pointV[0]->longitude() : pointV[1]->longitude();
+    double maxLon = (pointV[0]->longitude() > pointV[1]->longitude())? pointV[0]->longitude() : pointV[1]->longitude();
+
+    // Lower left: smaller lat and long, upper right: greater lat and long
+    Point ll;
+    ll.latitudeSet(minLat);
+    ll.longitudeSet(minLon);
+    Point ur;
+    ur.latitudeSet(maxLat);
+    ur.longitudeSet(maxLon);
+    box.fill(&ll, &ur);
+
```

Comment:
Yes, that's the point. As long as the point are different ant not aligned, any combination will work.